org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -147,17 +145,39 @@ private String getMessage(IStatus status) {
 	}
 
 	private class ServerStatusMonitor extends NullProgressMonitor{
+		private final static long DELAY = 300;
+
 		private double totalWork;
+		private String subtask;
 		private double progress;
+		private long lastReport = 0;
+
 		@Override
-		public void beginTask(String arg0, int totalWork) {
+		public void beginTask(String task, int totalWork) {
 			this.totalWork = totalWork;
+			sendProgress();
+		}
+
+		@Override
+		public void subTask(String name) {
+			this.subtask = name;
+			sendProgress();
 		}
 
 		@Override
 		public void worked(int work) {
 			progress += work;
-			connection.sendStatus(ServiceStatus.Starting,  String.format( "%.0f%%", progress/totalWork * 100));
+			sendProgress();
+		}
+
+		private void sendProgress() {
+			// throttle the sending of progress
+			long currentTime = System.currentTimeMillis();
+			if (currentTime - lastReport > DELAY) {
```

Comment:
so if the 100% status is skipped here, maybe done() should ensure the last progress is sent